src/NUnitEngine/nunit.engine.tests/Drivers/NUnit3FrameworkDriverTests.cs

```diff
@@ -149,11 +147,23 @@ public void RunTestsAction_AfterLoad_ReturnsRunnableSuite()
         public void RunTestsAction_WithoutLoad_ThrowsInvalidOperationException()
         {
             var ex = Assert.Catch(() => _driver.Run(new NullListener(), TestFilter.Empty.Text));
-            if (ex is System.Reflection.TargetInvocationException)
+            if (ex is TargetInvocationException)
                 ex = ex.InnerException;
             Assert.That(ex, Is.TypeOf<InvalidOperationException>());
             Assert.That(ex.Message, Is.EqualTo(LOAD_MESSAGE));
         }
+
+        [Test]
+        public void RunTestsAction_WithInvalidFilterElement_ThrowsNUnitEngineException()
+        {
+            _driver.Load(_mockAssemblyPath, _settings);
+
+            var invalidFilter = "<filter><invalidElement>foo</invalidElement></filter>";
+            var ex = Assert.Catch(() => _driver.Run(new NullListener(), invalidFilter));
+            if (ex is TargetInvocationException)
```

Comment:
TargetInvocationException means we are returning from a call through reflection. I guess that's because we are in the constructor called by the framework driver through reflection. However, if NUnit3FrameworkDriver catches those and wraps the original as an NUnitEngineException with it's own message, we should no longer see the TargetInvocationException.